packages/fuels/tests/test_projects/generics/src/main.sw

```diff
@@ -0,0 +1,108 @@
+contract;
+
+use std::assert::assert;
+use std::hash::sha256;
+
+struct SimpleGeneric<T> {
+    single_generic_param: T,
+}
+
+struct PassTheGenericOn<K> {
+    one: SimpleGeneric<K>,
+}
+
+struct StructWArrayGeneric<L> {
+    a: [L; 2],
+}
+
+struct StructWTupleGeneric<M> {
+    a: (M, M),
+}
+
+enum EnumWGeneric<N> {
+    a: u64,
+    b: N,
+}
+
+struct MegaExample<T, U> {
+    a: ([U; 2], T),
+    b: ([EnumWGeneric<StructWTupleGeneric<StructWArrayGeneric<PassTheGenericOn<T>>>>; 1], u32),
```

Comment:
![mother-of-god](https://user-images.githubusercontent.com/3484025/189956849-b5052c7d-2f6a-4d61-a646-5fd6276d8bd8.jpeg)